lib/solid/context.ex

```diff
@@ -14,17 +14,16 @@ defmodule Solid.Context do
 
   Possible scope values: :counter_vars, :vars or :iteration_vars
   """
-  @spec get_in(t(), [term()], [scope]) :: term
+  @spec get_in(t(), [term()], [scope]) :: {:ok, term} | {:error, :not_found, [key: [term()]]}
   def get_in(context, key, scopes) do
-    {:ok, result} =
-      scopes
-      |> Enum.map(&get_from_scope(context, &1, key))
-      |> Enum.find({:ok, nil}, fn
-        {:ok, _} -> true
-        _ -> false
-      end)
-
-    result
+    scopes
```

Comment:
I'd love to clean this up, but the case of `{:ok, nil}` really tripped me up. A nil value is a valid variable value, and I need to discern between it and incorrect key path